internal/resources/gitrepository/resource_git_repository.go

```diff
@@ -17,26 +29,309 @@ import (
+	if err != nil {
+		if !clienterrors.IsNotFoundError(err) {
+			return diag.FromErr(err)
+		}
+
+		err = enableContinuousDelivery(&config, scopedFullnameData, common.ConstructMeta(d))
+		if err != nil {
+			return diag.FromErr(errors.Wrapf(err, "Unable to create Tanzu Mission Control git repository entry, name : %s", gitRepositoryName))
+		}
+
+		return resourceGitRepositoryCreate(ctx, d, m)
+	}
+
+	var updateAvailable bool
+
+	if updateCheckForMeta(d, meta) {
+		updateAvailable = true
+	}
+
+	if updateCheckForSpec(d, atomicSpec, scopedFullnameData.Scope) {
+		updateAvailable = true
+	}
+
+	if !updateAvailable {
+		log.Printf("[INFO] git repository update is not required")
+		return
+	}
+
+	switch scopedFullnameData.Scope {
+	case commonscope.ClusterScope:
+		if scopedFullnameData.FullnameCluster != nil {
+			gitRepositoryReq := &gitrepositoryclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceFluxcdGitrepositoryGitRepositoryRequest{
+				GitRepository: &gitrepositoryclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceFluxcdGitrepositoryGitRepository{
+					FullName: scopedFullnameData.FullnameCluster,
+					Meta:     meta,
+					Spec:     atomicSpec,
+				},
+			}
+
+			_, err = config.TMCConnection.ClusterGitRepositoryResourceService.VmwareTanzuManageV1alpha1ClusterFluxcdGitrepositoryResourceServiceUpdate(gitRepositoryReq)
+			if err != nil && !clienterrors.IsFeatureDisabledError(err) {
+				return diag.FromErr(errors.Wrapf(err, "Unable to update Tanzu Mission Control cluster git repository entry, name : %s", gitRepositoryName))
+			}
+		}
+	case commonscope.ClusterGroupScope:
+		if scopedFullnameData.FullnameClusterGroup != nil {
+			gitRepositoryReq := &gitrepositoryclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupNamespaceFluxcdGitrepositoryGitRepositoryRequest{
+				GitRepository: &gitrepositoryclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupNamespaceFluxcdGitrepositoryGitRepository{
+					FullName: scopedFullnameData.FullnameClusterGroup,
+					Meta:     meta,
+					Spec: &gitrepositoryclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupNamespaceFluxcdGitrepositorySpec{
+						AtomicSpec: atomicSpec,
+					},
+				},
+			}
+
+			_, err = config.TMCConnection.ClusterGroupGitRepositoryResourceService.VmwareTanzuManageV1alpha1ClustergroupFluxcdGitrepositoryResourceServiceUpdate(gitRepositoryReq)
+			if err != nil && !clienterrors.IsFeatureDisabledError(err) {
+				return diag.FromErr(errors.Wrapf(err, "Unable to update Tanzu Mission Control cluster group git repository entry, name : %s", gitRepositoryName))
+			}
+		}
+	case commonscope.UnknownScope:
+		return diag.Errorf("no valid scope type block found: minimum one valid scope type block is required among: %v. Please check the schema.", strings.Join(scope.ScopesAllowed[:], `, `))
+	}
+
+	if err != nil {
+		err = enableContinuousDelivery(&config, scopedFullnameData, common.ConstructMeta(d))
```

Comment:
check this logic, 
I think this block should be removed and in the Update handler, we should only update the resource or justify why we are trying to create here ?

Reply:
Updated.